### PR TITLE
Scoreboards

### PR DESCRIPTION
**Pull request by Aaron1011**
This PR continues SpaceManiac's `scoreboard` branch, updaing it for 1.8

Working so far:
- Serialization/deserialization to/from [NBT scoreboard format](http://minecraft.gamepedia.com/Scoreboard#NBT_format).
- Teams
- Player scores
- Display slots - sidebar, below name, player list

In progress:
- Team color display slots (`sidebar.team.<color>`)
- Friendly fire toggling (Glowstone doesn't have damage
- Friendly invisibles
- Objective criterion (trigger, deathCount, playerKillCount, totalKillCount, health)
- Statistic/achievement criteria

### CODE REVIEW

**Comment by dequis** [comment]
Ok sooooo, what about #331? Did you guys talk?

**Comment by dequis** [comment]
Oh, found the answer scrolling up on irc.

```
19:21 < Aaron1011> turt2live: I've fully implemented serialization and deserialization now
19:21 <+turt2live> Neat. Open a PR when ready
19:22 < Aaron1011> turt2live: Okay, will do. Is this going to be competing with the other
                   scoreboard PR?
19:22 <+turt2live> Yes
19:22 <+turt2live> Let the odds ever be in your favour. 
19:23 < Aaron1011> :)
19:23 < Aaron1011> Let Tte 49th PR games begin!
```

Fun.

**Comment by Aaron1011** [comment]
@dequis: Yeah, it seems @turt2live wants some (friendly) PR competition :grinning:

File: src/main/java/net/glowstone/net/codec/play/scoreboard/ScoreboardScoreCodec.java [hunk 1]
```diff
@@ -15,10 +15,10 @@ public ScoreboardScoreMessage decode(ByteBuf buf) throws IOException {
     public ByteBuf encode(ByteBuf buf, ScoreboardScoreMessage message) throws IOException {
         final boolean remove = message.isRemove();
         ByteBufUtils.writeUTF8(buf, message.getTarget());
-        buf.writeBoolean(remove);
+        buf.writeByte(remove ? 1 : 0);
```
Comment (SpaceManiac) [inline src/main/java/net/glowstone/net/codec/play/scoreboard/ScoreboardScoreCodec.java:18]: Why this change?

**Comment by SpaceManiac** [comment]
I don't see where `Criteria` is coming from. Missing link to corresponding Glowkit PR?

**Comment by Aaron1011** [comment]
@SpaceManiac: Oops, forgot to create the other PR :).
https://github.com/GlowstoneMC/Glowkit/pull/20

**Comment by gdude2002** [comment]
This merges fine for me, I'm not sure why GitHub can't merge it. However, there are Checkstyle issues: http://bamboo.gserv.me/browse/GLOW-SRV-PR340-47

**Comment by dequis** [comment]
Because it's a PR against the 'scoreboard' branch, not 'master'. According to your build log there's no conflict with master.

**Comment by gdude2002** [comment]
@dequis Thanks, didn't see that. Updated my build script. Still merges, though.

Broken: http://bamboo.gserv.me/browse/GLOW-SRV-PR340-51

**Comment by gdude2002** [comment]
It lives! http://bamboo.gserv.me/browse/GLOW-SRV-PR340-51

**Comment by turt2live** [comment]
This is looking pretty inactive code-wise. If this is complete, please remove "WIP" from the title to have it reviewed. Your PR also does not merge.

If this is still a work in progress, please comment back stating so. 

Thanks!

(PS: Remember to limit your scope. No need to implement all of scoreboards just yet. This PR looks pretty hefty in the size department)

**Comment by turt2live** [comment]
@Aaron1011 It still does not merge :(

**Comment by dequis** [comment]
He merged against master but this PR is against the scoreboard branch.

I recommend resubmitting!

**Comment by gdude2002** [comment]
Well, that could cause some problems ._.

**Comment by turt2live** [comment]
Closing in favour of https://github.com/GlowstoneMC/Glowstone/pull/459
